src/framework/uicomponents/qml/Muse/UiComponents/internal/StyledDropdownView.qml

```diff
@@ -235,25 +277,20 @@ DropdownView {
                 }
 
                 Keys.onShortcutOverride: function(event) {
-                    if (event.text === "") {
-                        event.accepted = false
-                        return
-                    }
 
-                    if (prv.itemIndexByFirstChar(event.text) > -1) {
-                        event.accepted = true
-                    }
-                }
-
-                Keys.onPressed: function(event) {
-                    if (event.text === "") {
-                        return
-                    }
+                    var typedChar = event.text.toLowerCase();
+                    if (typedChar.length === 1 && /^[a-z0-9]$/.test(typedChar)) {
+                        if (typeAheadStr.length > 0 && lastTypedChar === typedChar) {
+                            typeAheadStr = typedChar;
+                        } else {
+                            typeAheadStr += typedChar;
+                        }
```

Comment:
I've updated the description slightly in https://github.com/musescore/MuseScore/issues/16508.

Basically, you should always add the typed character to the buffer, but if `typeAheadSameChar` is `true` then you only actually want to search for a single character, even though you save multiple characters.

The logic should be like this:

```QML
if (typeAheadSameChar && typeAheadStr.length > 0 && typedChar !== typeAheadLastChar) {
    typeAheadSameChar = false
}

typeAheadStr += typedChar // add to buffer regardless of typeAheadSameChar
typeAheadLastChar = typedChar

if (typeAheadSameChar) {
    prv.focusNextMatch(typedChar, currentIndex + 1)  // look for one character starting on next item
} else {
    prv.focusNextMatch(typeAheadStr, currentIndex)   // look for entire string starting on current item
}
```